resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsWatchSnapshotSource.scala

```diff
@@ -58,10 +58,14 @@ private[spark] class ExecutorPodsWatchSnapshotSource(
       snapshotsStore.updatePod(pod)
     }
 
-    override def onClose(e: KubernetesClientException): Unit = {
+    override def onClose(e: WatcherException): Unit = {
       logWarning("Kubernetes client has been closed (this is expected if the application is" +
```

Comment:
> As I see the current log message is misleading. If this method with the exception is called then "(this is expected if the application is shutting down.)" part is not true. So I think we should at least change that.

I guess @attilapiros comment is we are better to revise this log message? As this is exception case, it may not "this is expected".

Reply:
Actually, that's what @attilapiros meant, but it's not in reality. As of now, it's frequently possible to observe K8s side exceptions, especially HTTP-related ones, during application termination after `SparkContext.close()` starts to doing several clean-ups.

Reply:
Although `expected` was there, the message doesn't mean that `Exception will happen always`. Instead, it means you may see `Exception` some cases. In this context of `onClose`, `WatcherException` already happened. So, `logWarning` says the users that this exception is not **unexpected** one.

Reply:
BTW, @attilapiros and @viirya . If that is still your concerning, I'd suggest to handle it in a separate PR. That looks pretty much orthogonal to `K8s client 5.3.0`. WDYT?

Reply:
BTW, in this PR, the warning message of `onClose()` (without exception args) doesn't have `(this is expected if the application is...)` part.

Reply:
> Actually, that's what @attilapiros meant, but it's not in reality. As of now, it's frequently possible to observe K8s side exceptions, especially HTTP-related ones, during application termination after `SparkContext.close()` starts to doing several clean-ups.

In this case I am fine to leave it as it is.